cmd/run.go

```diff
@@ -383,6 +352,144 @@ func Run(isDebug *bool) *cli.Command {
 	}
 }
 
+func ReadState(fs afero.Fs, statePath string, filter *Filter) (*scheduler.PipelineState, error) {
+	pipelineState, err := scheduler.ReadState(fs, statePath)
+	if err != nil {
+		errorPrinter.Printf("Failed to restore state: %v\n", err)
+		return nil, err
+	}
+
+	filter.IncludeTag = pipelineState.Parameters.Tag
+
+	filter.OnlyTaskTypes = pipelineState.Parameters.Only
+
+	filter.PushMetaData = pipelineState.Parameters.PushMetadata
+
+	filter.ExcludeTag = pipelineState.Parameters.ExcludeTag
+
+	return pipelineState, nil
+}
+
+func GetPipeline(inputPath string, runConfig *scheduler.RunConfig, logger *zap.SugaredLogger) (*pipeline.Pipeline, bool, bool, *config.Config, error) {
```

Comment:
can you please simplify the return values from this function? it seems to be returning multiple bools, maybe we should be more explicit about what is being returned. maybe return a struct?